pkg/labels/labels.go

```diff
@@ -164,6 +164,14 @@ func (l Labels) String() string {
 	return str
 }
 
+// AddFromLabelArray adds the labels in a LabelArray keyed by each label's
+// String representation
+func (l Labels) AddFromLabelArray(lbls LabelArray) {
+	for _, lbl := range lbls {
+		l[lbl.String()] = lbl
```

Comment:
This is not respecting the documentation of the Labels struct:

https://github.com/cilium/cilium/blob/5fc5b37a2b9d31da0bef6ea61eaac63f7fdf24ff/pkg/labels/labels.go#L151-L152


The map's key should be the same as the labels' key.